### PR TITLE
add defaults, use url instead of provider, tests improvements

### PR DESCRIPTION
**Pull request by JustynaBroniszewska**

### CODE REVIEW

File: universal-login-sdk/lib/config.js [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
   gasPrice: 1000000000,
   gasLimit: 1000000
 };
 
-export default DEFAULT_PAYMENT_OPTIONS;
+const DEFAULTS = {
+  operationType: OPERATION_CALL,
+  value: 0,
+  data: '0x0'
+};
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/config.js]: Why not have just one`DEFAULTS`, why split into `DEFAULTS` and `DEFAULT_PAYMENT_OPTIONS`.

btw naming: `MESSAGE_DEFAULTS`

File: universal-login-sdk/test/fixtures/basicSDK.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  const relayer = await RelayerUnderTest.createPreconfigured(provider);
+  await relayer.start();
+  ({provider} = relayer);
+  const sdk = new EthereumIdentitySDK(relayer.url(), provider);
+  const [privateKey, identityAddress] = await sdk.create('alex.mylogin.eth');
+  const mockToken = await deployContract(wallet, MockToken);
+  await mockToken.transfer(identityAddress, utils.parseEther('1.0'));
+  await wallet.send(identityAddress, utils.parseEther('1.0'));
+  return {wallet, provider, mockToken, otherWallet, otherWallet2, sdk, privateKey, identityAddress, relayer};
+}
```
Comment (marekkirejczyk) [inline universal-login-sdk/test/fixtures/basicSDK.js]: +1

File: universal-login-sdk/test/sdk.js [hunk 3]
```diff
@@ -75,58 +50,44 @@ describe('SDK - integration', async () => {
...
-          value: 10,
-          data: utils.hexlify(0),
-          gasToken: token.address,
-          gasPrice,
-          gasLimit: utils.parseEther('25').toString() 
-        };
-        await expect(sdk.execute(identityAddress, message, privateKey)).to.be.eventually.rejected;
+      it('when not enough tokens ', async () => {
+        message = {...transferMessage, from: identityAddress, gasLimit: utils.parseEther('25').toString()};
+        await expect(sdk.execute(message, privateKey)).to.be.eventually.rejected;
```
Comment (marekkirejczyk) [inline universal-login-sdk/test/sdk.js]: Why not `rejectedWith`?

File: universal-login-sdk/lib/sdk.js [hunk 4]
```diff
@@ -93,14 +89,17 @@ class EthereumIdentitySDK {
...
-  async execute(contractAddress, message, privateKey) {
+  async execute(message, privateKey) {
     const url = `${this.relayerUrl}/identity/execution`;
     const method = 'POST';
-    const wallet = new ethers.Wallet(privateKey, this.provider);
-    const nonce = parseInt(await this.getNonce(contractAddress, wallet), 10);
-    message.nonce = nonce;
-    const signature = calculateMessageSignature(privateKey, {...message, from: contractAddress});
-    const body = JSON.stringify({...this.defaultPaymentOptions, ...message, contractAddress, signature});
+    const msg = {
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/sdk.js]: such an ugly short... why not `finalMessage`

Comment (JustynaBroniszewska) [reply universal-login-sdk/lib/sdk.js]: ✅

File: universal-login-sdk/test/sdk.js (hunk 3 above)
Comment (JustynaBroniszewska) [reply universal-login-sdk/test/sdk.js]: ✅

File: universal-login-sdk/test/fixtures/basicSDK.js (hunk 2 above)
Comment (JustynaBroniszewska) [reply universal-login-sdk/test/fixtures/basicSDK.js]: ✅

File: universal-login-sdk/lib/config.js (hunk 1 above)
Comment (JustynaBroniszewska) [reply universal-login-sdk/lib/config.js]: ✅

**Pull request merged**